clisqlclient: trim down dependencies

knz opened the pull request:
Fixes #67482.
Informs  #67470.

knz committed:
clisqliclient: drop the dependency on `roachpb`

Importing `roachpb` imports a bunch of server code into the SQL CLI
for no good reason.

This patch removes the dependency by using int32 in `clisqlclient`.
(The caller code in `cli` still uses `roachpb`, but that's all right.)

Release note: None

knz committed:
clisqlclient: stop depending on `tree.Name`

The code was using string concatenation with a `tree.Name`-based
escaped string, for no good reason. We can pass the current database
as a placeholder value instead.

Release note: None

knz committed:
clisqlclient: drop the dependency on `tree`

This patch removes the dependency `clisqlclient` -> `sql/sem/tree`, to
ensure that we don't depend on server-side functionality in the SQL
connection code.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/67489)
<!-- Reviewable:end -->

otan commented on pkg/cli/clisqlclient/string_to_duration.go:
```diff
@@ -0,0 +1,53 @@
+// depending on package `tree`, which makes the SQL shell executable
+// significantly larger.
+func stringToDuration(s string) (time.Duration, error) {
+	// The format is HH:MM:SS[.ffffff]
```
this will fail when the HH:MM:SS is in the magnitude of days

otan replied:
i'd extract the parseDuration in tree to the duration package in this case.

knz replied:
The interval style thing is a problem.

As you've noticed, the CLI code was hardwiring the interval style to POSTGRES. This was nonsense IMHO, because if the user changes the interval style server-side, the CLI code would balk already.

I don't think I want to fix bugs in this PRs that existed already, so I'm not going to do anything about the interval style here. 

I will try to extend the hand-written parser here to support days, however I'd like to point out that we have various hard timeouts inside CockroachDB that prevent queries from running for more than an hour, so that's that. 

The real fix would be to make the SHOW statement returns seconds (as floats) and call it a day. I'll file a separate issue for that.

knz replied:
Filed #67618 for  the core issue

knz replied:
I'll move forward as discussed here: https://github.com/cockroachdb/cockroach/issues/67618#issuecomment-880113168

otan replied:
> As you've noticed, the CLI code was hardwiring the interval style to POSTGRES. This was nonsense IMHO, because if the user changes the interval style server-side, the CLI code would balk already.

It's not a bug because at parse time, there's only a difference with how ParseDInterval works is how negative values work. Timings are never negative. This is captured in that comment.

This change here will become the regression -- ISO8601 and days won't work.

knz commented:
I have updated the last commit as discussed. RFAL

cc @arulajmani

knz committed:
clisqlclient: drop the dependency on `parser`

This moves `HasMultipleStatements` to package `scanner` which is much
smaller, and uses that.

Release note: None

knz commented:
TFYR!

bors r=otan

knz commented:
bors r=otan

knz commented:
bors r-

knz commented:
bors r=otan
